### ISSUE TITLE
AMD-GPU drivers 20.40 not detected by RBM...Teamred standalone (for exemple); works

### ISSUE THREAD

**Issue by ArSd-g** [RainbowMiner/RainbowMiner#1565]

**Comment by ArSd-g** [comment]
Thats weird when i run "sudo ./install.sh" amdgpu are detected than ./start.sh it doesn"t, only cpu!

**Comment by ArSd-g** [comment]
???

**Comment by RainbowMiner** [comment]
I have missed that, sorry. Could you please run `./gputest.sh` and post the result? I suspect, that there is no OpenCL coming with this driver.

**Comment by ArSd-g** [comment]
Hi, no problem...
Here it is:
laarsn@arsrl:~/RainbowMiner$ ./gputest.sh 
WARNING: Could not execute nvidia-smi --query-gpu=gpu_name,utilization.gpu,utilization.memory,temperature.gpu,power.d
raw,power.limit,fan.speed,pstate,clocks.current.graphics,clocks.current.memory,power.max_limit,power.default_limit --
format=csv,noheader: Exception calling "Start" with "0" argument(s): "No such file or directory"
MethodInvocationException: /home/laarsn/RainbowMiner/Scripts/GPUtest.ps1:106
Line |
 106 |      [OpenCl.Platform]::GetPlatformIDs() | ForEach-Object {
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "GetPlatformIDs" with "0" argument(s): "Value cannot be null. (Parameter 'path1')"

Done! Now please drop the file
 
/home/laarsn/RainbowMiner/gputestresult.txt
 
onto your issue at https://github.com/RainbowMiner/RainbowMiner/issues
 
Press any key to continuelaarsn@arsrl:~/RainbowMiner$ ^C
laarsn@arsrl:~/RainbowMiner$ sudo ./gputest.sh 
WARNING: Could not execute nvidia-smi --query-gpu=gpu_name,utilization.gpu,utilization.memory,temperature.gpu,power.d
raw,power.limit,fan.speed,pstate,clocks.current.graphics,clocks.current.memory,power.max_limit,power.default_limit --
format=csv,noheader: Exception calling "Start" with "0" argument(s): "No such file or directory"
MethodInvocationException: /home/laarsn/RainbowMiner/Scripts/GPUtest.ps1:106
Line |
 106 |      [OpenCl.Platform]::GetPlatformIDs() | ForEach-Object {
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "GetPlatformIDs" with "0" argument(s): "Value cannot be null. (Parameter 'path1')"

Done! Now please drop the file
 
/home/laarsn/RainbowMiner/gputestresult.txt
 
onto your issue at https://github.com/RainbowMiner/RainbowMiner/issues
 
Press any key to continue

Weird ... nothing about AMD but it is detected at the end of 'sudo /install.sh', teamredminer works very good with this driver (own opencl kernel?) should i go to 20.50? if i could stay >20.40 it would be great!

**Comment by RainbowMiner** [comment]
Another question, does `clinfo` find the GPUs?

**Comment by RainbowMiner** [comment]
Another pretty neat guide (this is for Debian linux):  https://gitlab.com/BCMM/amdgpu-opencl-on-debian

**Comment by RainbowMiner** [comment]
So it all comes down to testing with:
> Another question, does `clinfo` find the GPUs?

.. as soon, as the GPU(s) are listed by this tool, RainbowMiner should be able to pick them up.

**Comment by ArSd-g** [comment]
20.50=same

**Comment by RainbowMiner** [comment]
What does `clinfo` say? Are the 3 GPUs listed there? Our `install.sh` uses `lspci` for a basic check for GPUs, so that one sees every GPU, that is plugged into the PCIe ports. But - we do need to have them accessible via OpenCL for most of the miners. So please try `clinfo` at the command line (if it is not found, install it first `apt install clinfo`)

Regarding the install, please try not to mix the opencl parameter values. Try simple `--opencl=rocm` at first.

**Comment by RainbowMiner** [comment]
What you could try, is to install the mesa OpenCL. It did solve the Platform-not-found problem for some Ubuntu users.
`sudo apt install mesa-opencl-icd`

**Comment by RainbowMiner** [comment]
Here is a very interesting discussion plus solution and installer-script on how to successfully install AMD drivers with full OpenCL support:
https://gist.github.com/kytulendu/3351b5d0b4f947e19df36b1ea3c95cbe

**Comment by ArSd-g** [comment]
Great thanks for the infos..
("raw" no mesa installed  drivers-20.50 argument=rocr,legacy,rocm **this "rocr" replace "pal" in newers drivers**
also additionnal packages;
sudo apt install build-essential cmake git libuv1-dev libssl-dev libhwloc-dev libmicrohttpd-dev lm-sensors htop opencl-rocr-amdgpu-pro-dev clinfo
Seeing 20.50 clinfo result warn me; 20.40 was giving much more lines....rolling back 20.40
![Uploading Capture d’écran du 2021-05-17 19-41-30.png…]()

**Comment by ArSd-g** [comment]
Runnin clinfo as root with mesa installed
[clinfo.txt](https://github.com/RainbowMiner/RainbowMiner/files/6495777/clinfo.txt)

**Comment by RainbowMiner** [comment]
Yes, that `NULL platform behavior` is exactly what causes the trouble.

**Comment by ArSd-g** [comment]
"ERROR: using --no-dkms with a pro install is invalid."

**Comment by RainbowMiner** [comment]
This is an AMD-GPU-detection report against the amdgpu 20.40 Linux driver - that driver/distro combination is long end-of-life. The current AMD detection threads are #2758 and #3128 - please continue there, or open a fresh issue with a current debug file if a modern driver still fails.
